<?xml version="1.0"?>
<launch>

  <arg name="UAV_NAME" default="$(env UAV_NAME)"/>
  <arg name="config_drone_manager" default=""/>
  <arg name="config_trajectory_file_1" default=""/>
  <arg name="config_trajectory_file_2" default=""/>
  <arg name="frame_origin" default=""/>

  <group ns="$(arg UAV_NAME)">

    <node name="drone_manager" pkg="drone_manager" type="drone_manager_uav1.py" output="screen" >

      <!-- Load a user param file ## DRONE MANAGER -->
      <rosparam if="$(eval not arg('config_drone_manager') == '')" file="$(arg config_drone_manager)" />
      <rosparam unless="$(eval not arg('config_drone_manager') == '')" file="$(find drone_manager)/config/drone_manager.yaml" />

      <!-- Load a user param file -->
      <!-- 	TRAJECTORY_1 -->
      <param if="$(eval not arg('config_trajectory_file_1') == '')" name="trajectory_file_1" value="$(arg config_trajectory_file_1)" />
      <param unless="$(eval not arg('config_trajectory_file_1') == '')" name="trajectory_file_1" value="$(find drone_manager)/trajectories/test_traj.csv" />
      <!-- 	TRAJECTORY_2 -->
      <param if="$(eval not arg('config_trajectory_file_2') == '')" name="trajectory_file_2" value="$(arg config_trajectory_file_2)" />
      <param unless="$(eval not arg('config_trajectory_file_2') == '')" name="trajectory_file_2" value="$(find drone_manager)/trajectories/test_traj.csv" />

      <!-- Load a user param file ## REFERENCE FRAME -->
      <param name="reference_frame" value="$(arg UAV_NAME)/$(arg frame_origin)" />

      <!-- Subscribers -->
      <remap from="~control_manager_diagnostics_in" to="control_manager/diagnostics" />

      <!-- Services in -->
      <remap from="~start_in" to="~start" />

      <!-- Services out -->
      <remap from="~land_out" to="uav_manager/land" />
      <remap from="~takeoff_out" to="uav_manager/takeoff" />
      <remap from="~path_out" to="trajectory_generation/path" />
      <remap from="~goto_start_out" to="control_manager/goto_trajectory_start" />
      <remap from="~start_tracking_out" to="control_manager/start_trajectory_tracking" />
      <remap from="~trajectory_out" to="control_manager/trajectory_reference" />

    </node>

  </group>
</launch>
